android/app/src/androidTest/java/com/trustwallet/core/app/blockchains/everscale/TestEverscaleSigner.kt

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+package com.trustwallet.core.app.blockchains.everscale
+
+import com.google.protobuf.ByteString
+import com.trustwallet.core.app.utils.Numeric
+import com.trustwallet.core.app.utils.toHexByteArray
+import com.trustwallet.core.app.utils.toHexBytes
+import com.trustwallet.core.app.utils.toHexBytesInByteString
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import wallet.core.java.AnySigner
+import wallet.core.jni.CoinType.EVERSCALE
+import wallet.core.jni.proto.Everscale
+import wallet.core.jni.proto.Everscale.SigningOutput
+
+class TestEverscaleSigner {
+
+    init {
+        System.loadLibrary("TrustWalletCore")
+    }
+
+    @Test
+    fun testSign() {
+        val transferMessage = Everscale.Transfer.newBuilder().apply {
+            bounce = false
+            behavior = Everscale.MessageBehavior.SimpleTransfer
+            amount = 100000000
+            expiredAt = 1680770631
+            to = "0:db18a67f4626f15ac0537a18445937f685f9b30184f0d7b28be4bdeb92d2fd90"
+            encodedContractData = "te6ccgEBAQEAKgAAUAAAAAFLqS2KOWKN+7Y5OSiKhKisiw6t/h2ovvR3WbapyAtrdctwupw="
```

Comment:
Is it clear for the clients what to put here?

Reply:
`encodedContractData` will always be fetched from the blockchain before the transfer, with smth like this

```graphql
query {
  accounts(
    filter: { id: { eq: "0:df112b59eb82792623575194c60d2f547c68d54366644a3a5e02b8132f3c4c56" } }
  ) {
    data
  }
}

# Output:
# {
#   "data": {
#     "accounts": [{
#       "data": "te6ccgEBAQEAKgAAUAAAAANLqS2KOWKN+7Y5OSiKhKisiw6t/h2ovvR3WbapyAtrdctwupw="
#     }]
#   }
# }
```
[<sub>https://gra01.main.everos.dev/graphql</sub>](https://gra01.main.everos.dev/graphql)